src/hyperion/external_interaction/callbacks/ispyb_callback_base.py

```diff
@@ -77,41 +91,83 @@ def activity_gated_event(self, doc: Event) -> Event:
             )
             return doc
         if event_descriptor.get("name") == CONST.PLAN.ISPYB_HARDWARE_READ:
+            assert self._event_driven_data_collection_info
             ISPYB_LOGGER.info("ISPyB handler received event from read hardware")
-            self.params.hyperion_params.ispyb_params.undulator_gap = doc["data"][
+            self._event_driven_data_collection_info.undulator_gap1 = doc["data"][
                 "undulator_current_gap"
             ]
-            self.params.hyperion_params.ispyb_params.synchrotron_mode = doc["data"][
+            self._event_driven_data_collection_info.synchrotron_mode = doc["data"][
                 "synchrotron-synchrotron_mode"
             ]
-            self.params.hyperion_params.ispyb_params.slit_gap_size_x = doc["data"][
+            self._event_driven_data_collection_info.slitgap_horizontal = doc["data"][
                 "s4_slit_gaps_xgap"
             ]
-            self.params.hyperion_params.ispyb_params.slit_gap_size_y = doc["data"][
+            self._event_driven_data_collection_info.slitgap_vertical = doc["data"][
                 "s4_slit_gaps_ygap"
             ]
-            self.params.hyperion_params.ispyb_params.sample_barcode = doc["data"][
-                "robot-barcode"
-            ]
+            self._sample_barcode = doc["data"]["robot-barcode"]
 
         if event_descriptor.get("name") == CONST.PLAN.ISPYB_TRANSMISSION_FLUX_READ:
-            self.params.hyperion_params.ispyb_params.transmission_fraction = doc[
-                "data"
-            ]["attenuator_actual_transmission"]
-            self.params.hyperion_params.ispyb_params.flux = doc["data"][
+            assert self._event_driven_data_collection_info
+            if doc["data"]["attenuator_actual_transmission"]:
+                # Ispyb wants the transmission in a percentage, we use fractions
+                self._event_driven_data_collection_info.transmission = (
+                    doc["data"]["attenuator_actual_transmission"] * 100
+                )
+                # TODO 1173 Remove this once nexus_utils no longer needs it
+                self.params.hyperion_params.ispyb_params.transmission_fraction = doc[
+                    "data"
+                ]["attenuator_actual_transmission"]
```

Comment:
Should: I think this disappears on merge?

Reply:
No it doesn't but I will remove them